cli/command/command.go

```diff
@@ -221,6 +229,8 @@ func Populate(context *project.Context, c *cli.Context) {
 		context.Log = true
 	} else if c.Command.Name == "up" {
 		context.Log = !c.Bool("d")
+		context.NoRecreate = c.Bool("no-recreate")
+		context.ForceRecreate = c.Bool("force-recreate")
```

Comment:
I think these should be [converted to an enum](https://github.com/docker/compose/blob/f3d49e38c0e488194cd507b092111e0bce21e779/compose/cli/main.py#L622) and passed as a parameter to `project.Up()`. They aren't really part of the "project context", they're a parameter for a single operation.